### DISCUSSION TITLE
Delay in Reopening Sheet After Dismissal

### DISCUSSION THREAD

**Discussion by ammerzon** [pointfreeco/swift-composable-architecture#3197] [General]
### Description

I am unsure if this is a bug or standard SwiftUI behavior. It might be related to [issue #3045](https://github.com/pointfreeco/swift-composable-architecture/issues/3045).

Here's a simple example: on a home screen, when a button is clicked, a sheet opens. After dismissing the sheet, clicking the button again doesn't work immediately; there is a delay of a few seconds before it responds. I suspect the state hasn't changed to `nil`, preventing the sheet from opening immediately. Is this a known issue?

### Checklist

- [X] I have determined whether this bug is also reproducible in a vanilla SwiftUI project.
- [ ] If possible, I've reproduced the issue using the `main` branch of this package.
- [X] This issue hasn't been addressed in an [existing GitHub issue](https://github.com/pointfreeco/swift-composable-architecture/issues) or [discussion](https://github.com/pointfreeco/swift-composable-architecture/discussions).

### Expected behavior

The sheet opens the first time a button is clicked after dismissing the sheet.

### Actual behavior

The sheet can't be shown immediately again, but the user must wait a bit.

### Steps to reproduce

https://github.com/pointfreeco/swift-composable-architecture/assets/7319168/0ec58569-62bd-434f-9415-6b83b861a9d9

[Sheet Bug Project.zip](https://github.com/user-attachments/files/15926128/Sheet.Bug.Project.zip)

### The Composable Architecture version information

1.11.2

### Destination operating system

iOS 17.5

### Xcode version information

15.4

### Swift Compiler version information

```shell
swift-driver version: 1.90.11.1 Apple Swift version 5.10 (swiftlang-5.10.0.13 clang-1500.3.9.4)
Target: arm64-apple-macosx14.0
```

**Comment by mbrandonw** [comment]
Hi @ammerzon, this is just how SwiftUI works. If you were to recreate the demo in vanilla SwiftUI you would find that you cannot present a sheet until the existing sheet has fully dismissed. 

Since this isn't an issue with the library I am going to convert it to a discussion. Please feel free to continue the conversation over there!

**Reply by ammerzon** [reply]
Thank you for the information. Do you know whether this state change can be triggered earlier?

**Reply by mbrandonw** [reply]
I don't think it's just a matter of getting the state to change earlier. UIKit/SwiftUI also have the problem that if you try to present something while something else is presented it prints warnings to the console letting you know that isn't right.

I just think you really have no choice but to wait until the sheet finishes animating away before you can present something new.

But maybe asking this question on a more general SwiftUI forum will get you a better answer.
